ci/compute-matrix.jq

```diff
@@ -1,62 +1,89 @@
-  if
-    env.BUILD_TYPE == "pull-request"
-  then
-    "staging"
-  else
-    $x.IMAGE_REPO
-  end;
+include "ci/shared";
 
-def compute_tag_prefix($x):
-  if
-    env.BUILD_TYPE == "branch"
-  then
-    ""
-  else
-    $x.IMAGE_REPO + "-" + env.PR_NUM + "-"
-  end;
+# CI specific computations
+def compute_ci($x):
+  def compute_ci_arch($x):
+    ["amd64"] |
+    if
+      $x.CUDA_VER > "11.2.2" and
+      $x.LINUX_VER != "centos7"
+    then
+      . + ["arm64"]
+    else
+      .
+    end |
+    $x + {ARCHES: .};
 
-def compute_image_name($x):
-  compute_repo($x) as $repo |
-  compute_tag_prefix($x) as $tag_prefix |
-  "rapidsai/" + $repo + ":" + $tag_prefix + "cuda" + $x.CUDA_VER + "-" + $x.LINUX_VER + "-" + "py" + $x.PYTHON_VER |
-  $x + {IMAGE_NAME: .};
+  def compute_image_name($x):
+    compute_repo($x) as $repo |
+    compute_tag_prefix($x) as $tag_prefix |
+    "rapidsai/" + $repo + ":" + $tag_prefix + "cuda" + $x.CUDA_VER + "-" + $x.LINUX_VER + "-" + "py" + $x.PYTHON_VER |
+    $x + {IMAGE_NAME: .};
 
-# Checks the current entry to see if it matches the given exclude
-def matches($entry; $exclude):
-  all($exclude | to_entries | .[]; $entry[.key] == .value);
+  ($x.exclude // []) as $excludes |
+  $x | del(.exclude) |
+  keys_unsorted as $matrix_keys |
+  to_entries |
+  map(.value) |
+  [
+    combinations |
+    lists2dict($matrix_keys; .) |
+    filter_excludes(.; $excludes) |
+    compute_ci_arch(.) |
+    compute_image_name(.)
+  ] |
+  {include: .};
 
-# Checks the current entry to see if it matches any of the excludes.
-# If so, produce no output. Otherwise, output the entry.
-def filter_excludes($entry; $excludes):
-  select(any($excludes[]; matches($entry; .)) | not);
+# Wheels specific computations
+def compute_wheels($x):
+  def compute_wheels_arch($x):
+    ["amd64"] |
+    if
+      ["ubuntu18.04", "centos7"] | index($x.LINUX_VER) != null
+    then
+      .
+    else
+      . + ["arm64"]
+    end |
+    $x + {ARCHES: .};
 
-def lists2dict($keys; $values):
-  reduce range($keys | length) as $ind ({}; . + {($keys[$ind]): $values[$ind]});
+  def compute_manylinux_version($x):
+    if
+      ["ubuntu18.04", "ubuntu20.04"] | index($x.LINUX_VER) != null
+    then
+      "manylinux_2_31"
+    else
+      "manylinux_2_17"
+    end |
+    $x + {MANYLINUX_VER: .};
 
-def compute_matrix($input):
-  ($input.exclude // []) as $excludes |
-  $input | del(.exclude) |
+  def compute_image_name($x):
+    compute_repo($x) as $repo |
+    compute_tag_prefix($x) as $tag_prefix |
+    "rapidsai/" + $repo + ":" + $tag_prefix + "cuda" + $x.CUDA_VER + "-" + $x.LINUX_VER + "-" + "py" + $x.PYTHON_VER |
+    $x + {IMAGE_NAME: .};
+
+  ($x.exclude // []) as $excludes |
+  $x | del(.exclude) |
   keys_unsorted as $matrix_keys |
   to_entries |
   map(.value) |
   [
     combinations |
     lists2dict($matrix_keys; .) |
     filter_excludes(.; $excludes) |
-    compute_arch(.) |
+    compute_wheels_arch(.) |
+    compute_manylinux_version(.) |
```

Comment:
same comment about the `MANYLINUX_VER` here.

when you remove this `compute_manylinux_version` function, then the only difference between `compute_ci` and `compute_wheels` is how the architecture is computed.

so then you can update the function that computes the architecture to accommodate the logic for wheels and conda image variants and then do the following:

- revert the changes made to the `compute_matrix` function in this file
- remove `shared.jq` and put those functions back in this file